File: 프로그래머스/0/181916. 주사위 게임 3/주사위 게임 3.swift
import Foundation

func solution(_ a:Int, _ b:Int, _ c:Int, _ d:Int) -> Int {
    var s: Set<Int> = []
    let arr: [Int] = [a, b, c, d].sorted()
    arr.forEach { s.insert($0) }
    let cnt = s.count
    if cnt == 1 {
        return 1111 * arr[0]
    } else if cnt == 2 {
        if arr[1] == arr[2] {
            if arr[1] == arr[0] {
                return (10 * arr[0] + arr[3]) * (10 * arr[0] + arr[3])
            } else {
                return (10 * arr[3] + arr[0]) * (10 * arr[3] + arr[0])
            }
        } else {
            return (arr[1] + arr[2]) * (abs(arr[1] - arr[2]))
        }
    } else if cnt == 3 {
        if arr[0] == arr[1] {
            return arr[2] * arr[3]
        } else if arr[1] == arr[2] {
            return arr[0] * arr[3]
        } else  {
            return arr[0] * arr[1]
        }
    } else {
        return arr[0]
    }
}

File: 프로그래머스/0/181916. 주사위 게임 3/주사위 게임 3.cpp
#include <algorithm>

using namespace std;

int solution(int a, int b, int c, int d) {
    vector<int> v = {a, b, c, d};
    
    sort(v.begin(), v.end());
    if(v[0] == v[3]) return 1111 * v[0];
    else if(v[0] == v[2]) return (10 * v[0] + v[3]) * (10 * v[0] + v[3]);
    else if(v[1] == v[3]) return (10 * v[1] + v[0]) * (10 * v[1] + v[0]);
    else if(v[0] == v[1]) {
        if(v[2] == v[3]) return (v[0] + v[2]) * abs(v[0] - v[2]);
        else return v[2] * v[3];
    }
    else if(v[1] == v[2]) {
        if(v[0] == v[3]) return (v[0] + v[1]) * abs(v[0] - v[1]);
        else return v[0] * v[3];
    }
    else if(v[2] == v[3]) {
        if(v[0] == v[1]) return (v[0] + v[2]) * abs(v[0] - v[2]);
        else return v[0] * v[1];
    }
    else return v[0];
}